### PR TITLE
fix: issue with tabIndex for keyboard navigation

### LINKED ISSUE #1054: Keyboard accessibility not working
I am missing browser page navigation by the Pos1/End/PgUp/PgDn keys for a long time. I am used to these keys since they work on virtually every page in the browser.
Would you mind supporting/not disabling these keys to make navigation in ct documentation a more fluent experience?

### PR DESCRIPTION
**Pull request by emmenko**
Attempt to fix #1054

### CODE REVIEW

File: packages/gatsby-theme-docs/src/components/search-input.js [hunk 1]
```diff
@@ -77,6 +77,7 @@ const SearchInput = React.forwardRef((props, ref) => {
   const handleBlur = () => {
     if (props.isDisabled) return;
     setIsActive(false);
+    props.onClose();
```
Comment (emmenko) [inline packages/gatsby-theme-docs/src/components/search-input.js:80]: Unrelated to the original issue, but when using tab navigation for the search input, when the input blurs the search dialog should close as it's not the focused element anymore.

**Comment by emmenko** [comment]
So testing the deploy preview on Chrome it seems to work. On FF and Safari not really, the tab navigation is also almost nonexistent compared to Chrome.

File: packages/gatsby-theme-docs/src/layouts/internals/layout-main.js [hunk 2]
```diff
@@ -26,7 +26,13 @@ const Container = styled.main`
 `;
 
 const LayoutMain = (props) => (
-  <Container preventScroll={props.preventScroll}>
+  <Container
+    preventScroll={props.preventScroll}
+    // This is important to make the main container the active element when the page loads.
+    // The user can then use the keyboard navigation for example to scroll up/down.
+    autoFocus
```
Comment (nkuehn) [inline packages/gatsby-theme-docs/src/layouts/internals/layout-main.js:33]: Works in chrome but not firefox.  It seems one still has to call `node.focus()` in javascript to focus a specific node in firefox.

**Review by nkuehn — COMMENTED** [review]
Thanks for the improvement!  

If we want consistent behavior it requires a little more work though.  The preview of this PR only works on the initial page load, once you navigate inside the SPA the focus is not happening any more (I guess that's the intended behavior of the HTML level `autoFocus` - evaluated on HTML level page load but not after arbitrary javascript executions)

Since Firefox requires a script-based  `focus()` call too we might want to evaluation that approach?

**Comment by emmenko** [comment]
I think I found some useful information on how to address this in Gastby: https://www.gatsbyjs.com/blog/2020-02-10-accessible-client-side-routing-improvements/

Let me give it a try.

**Comment by emmenko** [comment]
Ok, this seems to work better now. @nkuehn @jenschude would you mind doing a quick manual test in the deploy link? Thanks

**Comment by jenschude** [comment]
It seems to be better now. At least I can now directly navigate inside the page with up/down after clicking a link.

**Review by nkuehn — APPROVED** [review]
nice, thank you!  It works now in the (Macos only) environments and use case that I have available.

**Pull request merged**